manual/cypher/cypher-docs/src/docs/dev/syntax/parameters.asciidoc

```diff
@@ -3,19 +3,32 @@
 
 Cypher supports querying with parameters.
 This means developers don't have to resort to string building to create a query.
-In addition to that, it also makes caching of execution plans much easier for Cypher.
+Additionally, parameters make caching of execution plans much easier for Cypher, thus leading to faster query execution times.
 
-Parameters can be used for literals and expressions in the `WHERE` clause, for the index value in the `START` clause, index queries, and finally for node/relationship ids.
-Parameters can not be used as for property names, relationship types and labels, since these patterns are part of the query structure that is compiled into a query plan.
+Parameters can be used for:
 
-Accepted names for parameters are letters and numbers, and any combination of these.
+* Literals and expressions (such as in the `WHERE` clause)
+* Node and relationship ids
+* For legacy indexes only: index values and queries
+
+Parameters cannot be used for the following constructs, as these form part of the query structure that is compiled into a query plan:
+
+* Property keys; so, `MATCH (n) WHERE n.$param = 'something'` is invalid
+* Relationship types
+* Labels
+
+Parameters may consist of letters and numbers, and any combination of these, but cannot start with a number or a currency symbol.
 
 For details on using parameters via the Neo4j REST API, see <<rest-api-transactional>>.
 // For details on parameters when using the Neo4j embedded Java API, see <<tutorials-cypher-parameters-java>>.
 
-Below follows a comprehensive set of examples of parameter usage.
-The parameters are given as JSON here.
-Exactly how to submit them depends on the driver in use.
+We provide below a comprehensive list of examples of parameter usage.
+In these examples, parameters are given in JSON; the exact manner in which they are to be submitted depends upon the driver being used.
+
+[NOTE]
+====
+It is recommended that the new parameter syntax -- `$param` -- is used, as the old syntax (`{param}`) is deprecated and will be removed altogether in a later release.
```

Comment:
I also think the use of `-- ... --` as well as `( ... )` as grouping is a bit odd. I'd prefer something consistent like:

    It is recommended that the new parameter syntax `$param` is used, as the old syntax `{param}` is deprecated and will be removed altogether in a later release.